magda-web-client/src/UI/DataPreviewMap.js

```diff
@@ -0,0 +1,182 @@
+            isInitLoading : true,
+            isMapPreviewAvailable : false,
+            selectedDistribution : null
+        }
+        this.onIframeMessageReceived = this.onIframeMessageReceived.bind(this);
+        this.iframeRef = null;
+    }
+
+    createStateFromProps(props){
+        this.iframeRef = null;
+
+        const { identifier } = props.dataset;
+        if(identifier === '') {
+            this.setState({
+                isInitLoading : true,
+                isMapPreviewAvailable : false,
+                selectedDistribution : null
+            });
+            return;
+        }
+        const selectedDistribution = this.determineDistribution(props);
+        if(!selectedDistribution) {
+            this.setState({
+                isInitLoading : false,
+                isMapPreviewAvailable : false,
+                selectedDistribution : null
+            });
+            return;
+        }else {
+            this.setState({
+                isInitLoading : false,
+                isMapPreviewAvailable : true,
+                selectedDistribution : selectedDistribution
+            });
+        }
+    }
+
+    componentWillMount(){
+        this.createStateFromProps(this.props);
+    }
+
+    componentDidMount(){
+        window.addEventListener('message', this.onIframeMessageReceived);
+    }
+    
+    componentWillUnmount(){
+        window.removeEventListener("message", this.onIframeMessageReceived);
+    }
+
+    componentWillReceiveProps(props){
+        this.createStateFromProps(props);
+    }
+
+    createCatalogItemFromDistribution(){
+        return {
+            "initSources": [
+                {
+                    "catalog": [
+                        {
+                            "name": this.state.selectedDistribution.title,
+                            "type": "magda-item",
+                            "url": "/",
+                            "distributionId": this.state.selectedDistribution.identifier,
+                            "isEnabled": true,
+                            "zoomOnEnable": true
+                        }
+                    ],
+                    "baseMapName": "Positron (Light)"
+                }
+            ]
+        };
+    }
+
+    determineDistribution(props){
+        const distributions = props.dataset.distributions;
+        let dataSourcePreference = props.dataset.dataSourcePreference;
+        if(!dataSourcePreference || !dataSourcePreference.length) dataSourcePreference = defaultDataSourcePreference;
+        dataSourcePreference = dataSourcePreference.map(item => item.toLowerCase());
+        if(!distributions || !distributions.length) return null;
+        let selectedDis = null, perferenceOrder = -1;
+        distributions.filter(item=>(item.linkStatusAvailable && item.linkActive)||!item.linkStatusAvailable).forEach(dis => {
+            const format = dis.format.toLowerCase();
+            const disPerORder = dataSourcePreference.indexOf(format);
```

Comment:
I find this to be a super confusing name - what's a disPerORder? is it like distribution preference order?

Reply:
Good pickup~ It was supposed to be `disPerOrder`. I think I'd better change it to `distributionPerferenceOrder` to avoid confusion.